contrib/generate_precompile.jl

```diff
@@ -15,12 +15,17 @@ UP_ARROW = "\e[A"
 DOWN_ARROW = "\e[B"
 
 hardcoded_precompile_statements = """
-precompile(Tuple{typeof(Base.stale_cachefile), String, String})
-precompile(Tuple{typeof(push!), Set{Module}, Module})
-precompile(Tuple{typeof(push!), Set{Method}, Method})
-precompile(Tuple{typeof(push!), Set{Base.PkgId}, Base.PkgId})
-precompile(Tuple{typeof(setindex!), Dict{String,Base.PkgId}, Base.PkgId, String})
-precompile(Tuple{typeof(get!), Type{Vector{Function}}, Dict{Base.PkgId,Vector{Function}}, Base.PkgId})
+@assert precompile(Tuple{typeof(Base.stale_cachefile), String, String})
+@assert precompile(Tuple{typeof(push!), Set{Module}, Module})
+@assert precompile(Tuple{typeof(push!), Set{Method}, Method})
+@assert precompile(Tuple{typeof(push!), Set{Base.PkgId}, Base.PkgId})
+@assert precompile(Tuple{typeof(setindex!), Dict{String,Base.PkgId}, Base.PkgId, String})
+@assert precompile(Tuple{typeof(get!), Type{Vector{Function}}, Dict{Base.PkgId,Vector{Function}}, Base.PkgId})
+@assert precompile(Tuple{typeof(FileWatching.watch_file), String, Float64})
```

Comment:
What happens with these if you compile without FileWatching in the sysimg?

Reply:
Yes, the ones that are dependent on a certain stdlib existing should go into conditionals similar to line ~50.